src/Mod/TechDraw/App/DrawGeomHatch.cpp

```diff
@@ -388,31 +388,33 @@ std::vector<LineSet> DrawGeomHatch::getTrimmedLines(DrawViewPart* source,
 /* static */
 std::vector<TopoDS_Edge> DrawGeomHatch::makeEdgeOverlay(PATLineSpec hl, Bnd_Box b, double scale)
 {
+    constexpr double RightAngleDegrees{90.0};
+    constexpr double HalfCircleDegrees{180.0};
     std::vector<TopoDS_Edge> result;
 
     double minX, maxX, minY, maxY, minZ, maxZ;
     b.Get(minX, minY, minZ, maxX, maxY, maxZ);
     //make the overlay bigger to cover rotations. might need to be bigger than 2x.
-    double centerX = (minX + maxX) / 2.0;
     double widthX = maxX - minX;
-    minX = centerX - widthX;
-    maxX = centerX + widthX;
-    double centerY = (minY + maxY) / 2.0;
     double widthY = maxY - minY;
-    minY = centerY - widthY;
-    maxY = centerY + widthY;
+    double width = std::max(widthX, widthY);
+
+    double centerX = (minX + maxX) / 2;
+    minX = centerX - width;
+    maxX = centerX + width;
+    double centerY = (minY + maxY) / 2;
+    minY = centerY - width;
+    maxY = centerY + width;
```

Comment:
Just for reference, this is where the magic happens.
Instead of looking at x and y separately, this uses the bigger of the widths.
The function will create an oversized overlay, but as shown in the report that wasn't always the case. With this code it will.

Reply:
We create an oversize hatch, then take the common of the oversize hatch and the face to be hatched.